### PR TITLE
Bugfix/cam var names

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
## Pull Request Testing ##

- [x] Describe testing already performed for this Pull Request:</br>

All jobs were run as described in this PR and completed normally.

### This bugfix addresses the following:
- Rename reserved variables as needed (cyc=>vhr, CYC=>VHR, cycles=>vhours)
- Fix a bug that causes failed METplus jobs when processing with bad SPC outlook dbf files (this is a refinement of a fix included in the most recent cam PR)
- Fix handling of error case when a full day of ccpa or mrms precip data are missing
- Fix handling of error case when no SPC Outlook mask files are available for the valid period
- Change COMOUT paths in drivers to match those used in the EVS parallel

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

### (1) Set up jobs 
•	symlink the EVS_fix directory locally as "fix"
•	In all driver scripts, point all exports of "HOMEevs" to your EVS test directory
•	In all driver scripts, change COMIN to the parallel directory (`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver`)
•	Optional: In all stats drivers scripts, change "maillist" to point to your email address
### (2) Test Jobs
Use the following …
•	…to submit stats/precip: `qsub –v vhr=19,VDATE=$(date -d "-3 days" +"%Y%m%d")`
•	…to submit everything else: `qsub –v vhr=00 `
(_Note change to "vhr" in qsub commands_)

From your EVS testing directory, the following scripts should be run using the set up in step (1):
```
dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_href_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hrrr_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_namnest_precip_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_namnest_severe_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_radar_prep.sh
dev/drivers/scripts/prep/cam/jevs_cam_severe_prep.sh
```
```
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_snowfall_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_snowfall_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_snowfall_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_href_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_href_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_href_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_href_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_href_spcoutlook_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hrrr_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hrrr_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hrrr_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hrrr_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_hrrr_snowfall_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_namnest_grid2obs_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_namnest_precip_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_namnest_radar_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_namnest_severe_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_namnest_snowfall_stats.sh
dev/drivers/scripts/stats/cam/jevs_cam_nam_firewxnest_grid2obs_stats.sh
```
```
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_cape_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_cape_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_ctc_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_ctc_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_ecnt_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_grid2obs_ecnt_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_precip_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_precip_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_precip_spatial_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_snowfall_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_snowfall_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_spcoutlook_past31days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_href_spcoutlook_past90days_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_nam_firewxnest_grid2obs_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_grid2obs_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_headline_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_precip_plots.sh
dev/drivers/scripts/plots/cam/jevs_cam_snowfall_plots.sh
```
_[Total: 13 prep jobs, 31 stats jobs, 18 plots jobs]_

- [x] Has the code been checked to ensure that no errors occur during the execution? **Yes**

- [x] Do these updates/additions include sufficient testing updates? **Yes**

- [x] Please complete this pull request review by **11/17/2023**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @MarcelCaron-NOAA , I have cloned your fork and checked out branch `bugfix/cam_var_names`.

**Comment by ShelleyMelchior-NOAA** [comment]
I have completed the running of prep processing. 
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam/
total 48
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 14:37 ccpa.20231016
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 14:37 ccpa.20231017
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:07 hiresw.20231019
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:05 href.20231019
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:07 hrrr.20231019
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 14:57 mrms.20231013
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 14:37 mrms.20231016
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 14:37 mrms.20231017
drwxr-sr-x 4 shelley.melchior vpppg 4096 Oct 20 15:15 mrms.20231019
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:13 nam.20231019
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:06 spc_lsr.20231012
drwxr-sr-x 2 shelley.melchior vpppg 4096 Oct 20 15:06 spc_otlk.20231019
```
Should I be concerned that some of the mrms directories are empty? It is causing some problems in the log files. For example, see line 11886 of `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_precip_prep.o106879796`

All log files: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_*prep*/o*`

File: jobs/JEVS_CAM_PREP
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_CAM_PREP:1]: Line 115:
```
mkdir -p $COMOUT
```
Should it be moved down to line 125 and be adjusted to include a few other directories?
```
mkdir -p $COMOUT $COMOUTmrms $COMOUTlsr $COMOUTotlk
```
The COMOUTmrms dir for 20231016 wasn't getting created but I don't know if that is b/c of a lack of `mkdir -p` or b/c of missing data upstream.

**Comment by MarcelCaron-NOAA** [comment]
> Should I be concerned that some of the mrms directories are empty? It is causing some problems in the log files. For example, see line 11886 of `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_precip_prep.o106879796`

Yes, the issue I think is in the values fed to qsub -v
```
 13262     Submit_arguments = -v vhr=19,
 13263     VDATE=20231017 /lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/
 13264     forks/MC/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_precip_pre
 13265     p.sh
```
Only precip stats job submission commands should use those arguments.  For prep, this should instead be qsub -v vhr=00 /lfs/h2/...

However I pushed a minor fix that prevents a FATAL ERROR from occurring in this situation, since I suppose such an error shouldn't occur, like it is here, if some input data are missing.

**Comment by ShelleyMelchior-NOAA** [comment]
> Only precip stats job submission commands should use those arguments. For prep, this should instead be qsub -v vhr=00 /lfs/h2/... 
❗  ok ... I misunderstood that in the instructions.

File: jobs/JEVS_CAM_PREP
Comment (MarcelCaron-NOAA) [reply jobs/JEVS_CAM_PREP:1]: COMOUTmrms I believe is written during jevs_cam_radar_prep, which used VDATE=20231019.  So COMOUTmrms should be: 
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam/mrms.20231019`

And it looks like that populated correctly:
```
[marcel.caron@dlogin02 EVS] ls /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam/mrms.20231019/*
/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam/mrms.20231019/alaska:
MergedReflectivityQCComposite_00.50_20231019-000000.G091.nc  MergedReflectivityQCComposite_ENS_FREQ_20231019-000000.G091.nc  MergedReflectivityQCComposite_Prob_20231019-000000.G091.nc

/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam/mrms.20231019/conus:
EchoTop18_00.50_20231019-000000.G227.nc     EchoTop18_Prob_20231019-000000.G227.nc                       MergedReflectivityQCComposite_ENS_FREQ_20231019-000000.G227.nc
EchoTop18_ENS_FREQ_20231019-000000.G227.nc  MergedReflectivityQCComposite_00.50_20231019-000000.G227.nc  MergedReflectivityQCComposite_Prob_20231019-000000.G227.nc
```

The respective directories seem to be created in /scripts:
```
[marcel.caron@dlogin02 EVS] git grep "COMOUTmrms"
jobs/JEVS_CAM_PREP:export COMOUTmrms=${COMOUT}/mrms.${VDATE}
scripts/prep/cam/exevs_cam_radar_prep.sh:mkdir -p $COMOUTmrms
scripts/prep/cam/exevs_cam_radar_prep.sh:            mkdir -p $COMOUTmrms/${DOMAIN}
scripts/prep/cam/exevs_cam_radar_prep.sh:               cp -v $FILE $COMOUTmrms/${DOMAIN}
```
```
[marcel.caron@dlogin02 EVS] git grep "COMOUTlsr"
jobs/JEVS_CAM_PREP:   export COMOUTlsr=${COMOUT}/spc_lsr.${REP_DATE}
scripts/prep/cam/exevs_cam_severe_prep.sh:   mkdir -p $COMOUTlsr
scripts/prep/cam/exevs_cam_severe_prep.sh:      mkdir -p $COMOUTlsr
scripts/prep/cam/exevs_cam_severe_prep.sh:         cp -v $FILE $COMOUTlsr
```
```
[marcel.caron@dlogin02 EVS] git grep "COMOUTotlk"
jobs/JEVS_CAM_PREP:   export COMOUTotlk=${COMOUT}/spc_otlk.${OTLK_DATE}
scripts/prep/cam/exevs_cam_severe_prep.sh:         mkdir -p $COMOUTotlk
scripts/prep/cam/exevs_cam_severe_prep.sh:            cp -v $FILE $COMOUTotlk
```

I think it is possible to move the `mkdir -p` commands to the J-job without issue.  Do you think that would be better?

Comment (ShelleyMelchior-NOAA) [reply jobs/JEVS_CAM_PREP:1]: It is possible, but they are being made properly in the downstream exevs scripts, and that is fine. I have no problem w/ that. Thank you for that extra check for me.

I re-ran one of the precip prep jobs using `qsub -v vhr=00` and using `qsub -v vhr=90,VDATE=...` Both ran to completion and w/o error. I appreciate you fortifying the code w/ a -z test b/c of my mistake interpreting the PR instructions. 🤣

**Comment by ShelleyMelchior-NOAA** [comment]
The stats processing is underway and mostly complete. hrrr snowfall and href g2o are still processing.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231019` (grid2obs, radar)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/*.20231013` (severe)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231018` (snowfall)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231017` (precip)
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_*stats*.o10688*`

**Comment by ShelleyMelchior-NOAA** [comment]
Plots processing is underway. cam grid2obs, precip, and snowfall are still processing.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/cam/atmos.20231019` (cam)
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam/atmos.20231018` (href)
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam/atmos.20231019` (nam firewx)
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_*_plots.o1068*`

File: jobs/JEVS_CAM_PREP
Comment (MarcelCaron-NOAA) [reply jobs/JEVS_CAM_PREP:1]: Awesome!  It turns out this was a good test to find an error case I missed.

**Comment by MarcelCaron-NOAA** [comment]
severe, radar, and reruns of precip prep jobs completed normally.

Prep jobs have an OK from me!

**Comment by MarcelCaron-NOAA** [comment]
radar, severe, snowfall, and precip stats jobs completed successfully and output stats files appear normal.

For grid2obs, href is processing normally so far, nam_firewxnest has completed normally, and the other cam-det have completed with METplus errors.  Looking into those now.

**Comment by MarcelCaron-NOAA** [comment]
Re: the jevs_cam_*_grid2obs_stats jobs ...

Errors occurred when running PointStat on the SPC Outlook masks because there were no masks available for the valid period.  For these cases, the valid period contained no SPC Outlook regions due to low severe weather risk. 

@ShelleyMelchior-NOAA I tested and pushed a fix to handle such cases.  Do you mind removing the restart file (see below) and then re-running one of the grid2obs jobs?  To remove the restart file:
```
rm /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231019/hireswarw/grid2obs/restart/c00/completed_jobs.txt
```

**Comment by ShelleyMelchior-NOAA** [comment]
Sure thing, @MarcelCaron-NOAA . The re-run of `jevs_cam_hireswarw_grid2obs_stats.sh` is processing.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231019/hireswarw/grid2obs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_grid2obs_stats.o106897668`

**Comment by MarcelCaron-NOAA** [comment]
The recent jevs_cam_hireswarw_grid2obs_stats run completed successfully, with a normal output file.  jevs_cam_href_grid2obs_stats also seems to have completed successfully.

After the reruns, stats jobs have an OK from me!

**Comment by MarcelCaron-NOAA** [comment]
So far: href, nam-firewx, cam-precip, cam-snowfall, and cam-headline plots jobs have finished normally, and output files look similar to those produced in the parallel.  I'll give @BinbinZhou-NOAA and @PerryShafran-NOAA a chance to check and give the OK themselves if they prefer.

Otherwise: waiting on cam-grid2obs plots, but so far so good. I'll check again later

**Comment by ShelleyMelchior-NOAA** [comment]
And all cam plots have completed except for grid2obs.
```
106890288.dbqs01               shelley.melchi* dev             jevs_cam_grid2*   126901    4   512    --  10:00 R 02:26
```

**Comment by MarcelCaron-NOAA** [comment]
grid2obs plots jobs completed successfully and the output plots look normal.

Plots jobs have an OK from me! 

No other concerns about this PR from me.

**Comment by ShelleyMelchior-NOAA** [comment]
I concur. This PR is ready for merge. I will do the formal code review and approve this PR for merge.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All tests yield expected results. This PR is approved for merge.

**Pull request merged**